### PR TITLE
Add Loss::MeanSquaredError Function & iris example

### PR DESCRIPTION
**Pull request by naitoh**
Add Loss::MeanSquaredError Function
Update Utils::Array::force_array function

### CODE REVIEW

**Comment by naitoh** [comment]
I found a problem with this pull request. Please wait while we fix the problem. I'm sorry.

**Comment by naitoh** [comment]
I fixed the problem.
Please check this pull request.

**Comment by naitoh** [comment]
I fixed all problem.
Please check this pull request.

File: examples/iris.rb [hunk 1]
```diff
@@ -0,0 +1,103 @@
+require 'numo/narray'
+require 'chainer'
+require "datasets"
```
Comment (hatappi) [inline examples/iris.rb]: When run iris.rb, error occurs.
`require': cannot load such file -- datasets (LoadError)`

You need to add [red-datasets](https://github.com/red-data-tools/red-datasets) to gemspec

File: examples/iris.rb [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+class_name = y_class.uniq
+# y => [0, 0, 0, 0, ,,, 1, 1, ,,, ,2, 2]
+y = y_class.map{|s|
+  class_name.index(s)
+}
+
+# y_onehot => One-hot [[1.0, 0.0, 0.0], [1.0, 0.0, 0.0],,, [0.0, 1.0, 0.0], ,, [0.0, 0.0, 1.0]]
+y_onehot = y_class.map{|s|
+  i = class_name.index(s)
+  a = Array.new(class_name.length, 0.0)
```
Comment (hatappi) [inline examples/iris.rb]: `size`, `length` are mixed.
I would like to unite `length`

File: test/variable_test.rb [hunk 3]
```diff
@@ -63,4 +80,60 @@ def test_label_cpu(data)
     _setup(data)
     check_label(@label, false)
   end
+
+  def check_backward(inputs, intermediates, outputs, retain_grad)
+    for o in outputs
```
Comment (hatappi) [inline test/variable_test.rb]: red chainer offten use `each` methods.
What do you think?

**Comment by hatappi** [comment]
@naitoh
I comment this pr.
Please check comment 🙏

**Comment by kou** [comment]
(This comment isn't this PR scope.)

It seems that we can improve Red Datasets API for this use case. Let's think about it in somewhere.

File: examples/iris.rb (hunk 2 above)
Comment (naitoh) [reply examples/iris.rb]: Thank you for pointing out.
Fixed with 1f2a607.

I unified it to use the "size" method according to other processing.

**Comment by naitoh** [comment]
@hatappi 
> I comment this pr.

I fixed about all the contents pointed out.
Thank you!

**Comment by hatappi** [comment]
@naitoh 
Thank you for contributing!!
I'm very happy to add tests of existing methods

**Pull request merged**